src/Fable.Transforms/Rust/Fable2Rust.fs

```diff
@@ -3530,9 +3545,14 @@ module Util =
                     []
                 else
                     let ty =
-                        let genArgs = getEntityGenArgs ent |> transformGenArgs com ctx
+                        let genArgs = getEntityGenArgs ent
+                        let entRef = Fable.DeclaredType(entRef, getEntityGenArgs ent)
+                        let genArgs = genArgs |> transformGenArgs com ctx
```

Comment:
This change from one to three lines of code seems unnecessary, as the end result looks the same. Is it needed?

Reply:
See line 3553, which requires entRef to be pulled out.

I cannot activate this line because it breaks IEnumerable casting, but I believe this must be turned on as soon as possible to enable self references to be of the Lrc wrapper

Reply:
@alexswan10k I see, can you then rewrite it to:
```fs
let entTyp = Fable.DeclaredType(entRef, genArgs)
...
//maybeWrapInPtrTy com ctx entTyp ty
```

Reply:
will fix, give me a bit